File: src/io/read_xml_output_diffusion.jl
function read_xml_output_diffusion(file_path::String)
	file_stream::IOStream = open(file_path, "r")
	file_string::String = readstring(file_stream)
	close(file_stream)
			
	diagnostic_diffusion_coefficient_ratio::Float64 = read_xml_key(file_string, "diagnostic_diffusion_coefficient_ratio", Float64)
	t::Array{Float64, 1} = read_xml_key(file_string, "time", Array{Float64, 1})
	msd_x::Array{Float64, 1} = read_xml_key(file_string, "mean_square_displacement_x", Array{Float64, 1})
	msd_y::Array{Float64, 1} = read_xml_key(file_string, "mean_square_displacement_y", Array{Float64, 1})
	msd_z::Array{Float64, 1} = read_xml_key(file_string, "mean_square_displacement_z", Array{Float64, 1})
	t_exec::Float64 = read_xml_key(file_string, "execution_time", Float64)
	
	return (diagnostic_diffusion_coefficient_ratio, t, msd_x, msd_y, msd_z, t_exec)
end

File: tools/read_xml_output_diffusion.m
function [diagnostic_diffusion_coefficient_ratio, t, msd_x, msd_y, msd_z, t_exec_diffusion] = read_xml_output_diffusion(file_path)

file_string = fileread(file_path);

diagnostic_diffusion_coefficient_ratio = read_xml_key(file_string, 'diagnostic_diffusion_coefficient_ratio', 'scalar');
t = read_xml_key(file_string, 'time', 'array');
msd_x = read_xml_key(file_string, 'mean_square_displacement_x', 'array');
msd_y = read_xml_key(file_string, 'mean_square_displacement_y', 'array');
msd_z = read_xml_key(file_string, 'mean_square_displacement_z', 'array');
t_exec_diffusion = read_xml_key(file_string, 'execution_time', 'scalar');

end

